tools/test/audio/comp_run.sh

```diff
@@ -116,8 +116,16 @@ run_testbench ()
         # shellcheck disable=SC2086
         $VALGRIND_CMD $CMD
     else
+        set +e
         # shellcheck disable=SC2086
         $VALGRIND_CMD $CMD 2> "$FN_TRACE"
+        RETVAL=$?
+        if [ $RETVAL -ne 0 ]; then
+            echo ----------------------------------------------------------
+            cat "$FN_TRACE"
+            echo ----------------------------------------------------------
+            exit $RETVAL
+        fi
```

Comment:
I just realized I should to restore `set -e `after this error handling code. With run success there's still possibly something that could (green) fail.

Reply:
Here's how you should be able to kill all birds with one stone:
```shell
else
   $VALGRIND_CMD $CMD 2> "$FN_TRACE" || {
      local ret=$?
      echo ----------------------------------------------------------
      cat "$FN_TRACE"
      echo ----------------------------------------------------------
      return $ret # "exit" would be for something unexpected and catastrophic,
                  # not for a "regular" test failure
    }
```

Reply:
Thanks! I used he above code version, it works correctly for a FW error return and a valgrind issue those I tried to insert to code.